docs/src/guides/events.md

```diff
@@ -58,4 +58,54 @@ ConcurrentSim.Event 1
 
 julia> run(sim)
 Called back from ConcurrentSim.Event 1
-```
\ No newline at end of file
+```
+
+## Example usages of Event
+
+The simple mechanics outlined above provide a great flexibility in the way events can be used.
+
+One example for this is that events can be shared. They can be created by a process or outside of the context of a process. They can be passed to other processes and chained:
+
+```jldoctest
+using ResumableFunctions
+using ConcurrentSim
+
+mutable struct School
+  class_ends :: Event
+  pupil_procs :: Vector{Process}
+  bell_proc :: Process
+  function School(env::Simulation)
+    school = new()
+    school.class_ends = Event(env)
+    school.pupil_procs = Process[@process pupil(env, school) for i=1:3]
+    school.bell_proc = @process bell(env, school)
+    return school
+  end
+end
+
+@resumable function bell(env::Simulation, school::School)
+  for i=1:2
+    @yield timeout(env, 45.0)
+    succeed(school.class_ends)
+    #school.class_ends = Event(env) -- ?? orig SimJulia example somehow works
```

Comment:
this line?

Reply:
yes, in old version this line was ok, but ConcurrentSim.jl was failing. When I moved this line right after yield statement (line 103 in events.md) everything works fine. So you can remove this commented line if there are no reason to investigate further

Reply:
I think this actually introduced a bug: instead of all 3 pupils waiting on the same event, in the second round they all wait on different events, only one of which gets unlocked. I think I fixed that, although I would appreciate if you double check.

I also added a warning above the example to make sure that users know this is a low-level functionality they should not need. This is more easily done with Resource and Store.

Reply:
yes, i agree, my change was the bug. I forgot at that moment we have more pupil processes so resetting of event after yield does not look right. that also means i need to take a closer look on how succeed() implemented - it seems that you resetting event while it could still block some other yields. Or succeed() have something to guarantee that all pending yields will be processed before succeed() returns?

Looks like I also need to learn how to use Resource and Store.

General question: do you know any written text on web (or just the book/paper) where all this discrete simulation primitives and interactions are described in coherent fashion?

Reply:
Ben wrote this library to mimic SimPy. That could be a useful reference.